rfcs/0044-ckb-light-client/0044-ckb-light-client.md

```diff
@@ -0,0 +1,508 @@
+    difficulty_boundary:        Uint256,
+    // The sampled difficulties.
+    difficulties:               Uint256Vec,
+}
+
+table SendLastStateProof {
+    // If the block whose hash is sent from the client is on the chain, then
+    // returns its verifiable header; otherwise, returns the verifiable
+    // header for the tip block in the server.
+    last_header:                VerifiableHeader,
+    // The MMR proof for the chain root whose hash is in the last header.
+    // Be empty if the block hash sent from the client isn't on the chain.
+    proof:                      HeaderDigestVec,
+
+    // Verifiable headers for all sampled blocks.
+    headers:                    VerifiableHeaderVec,
+}
+
+table GetBlocksProof {
+    // Refer to `GetLastStateProof.last_hash`.
+    last_hash:                  Byte32,
+
+    // Block hashes for the blocks which require verifying.
+    block_hashes:               Byte32Vec,
+}
+
+table SendBlocksProof {
+    // Refer to `SendLastStateProof.last_header`.
+    last_header:                VerifiableHeader,
+    // Refer to `SendLastStateProof.proof`.
+    proof:                      HeaderDigestVec,
+
+    // Block headers for the blocks which require verifying.
+    headers:                    HeaderVec,
+
+    // Block hashes for the blocks which were not found.
+    missing_block_hashes:       Byte32Vec,
+}
+
+table GetTransactionsProof {
+    // Refer to `GetLastStateProof.last_hash`.
+    last_hash:                  Byte32,
+
+    // Transaction hashes for the transactions which require verifying.
+    tx_hashes:                  Byte32Vec,
+}
+
+table SendTransactionsProof {
+    // Refer to `SendLastStateProof.last_header`.
+    last_header:                VerifiableHeader,
+    // Refer to `SendLastStateProof.proof`.
+    proof:                      HeaderDigestVec,
+
+    // A collection of filtered blocks, which include all requested
+    // transactions, and be verified in the proof.
+    filtered_blocks:            FilteredBlockVec,
+
+    // Transaction hashes for the blocks which were not found.
+    missing_tx_hashes:          Byte32Vec,
+}
+```
+
+### Client-Server Interaction
+
+We can divide the typical client-server interaction into 3 phases.
+
+#### Phase #1: Client synchronizes to the latest tip block.
+
+- At start, the client send `GetLastState` messages to all discovered
+  servers.
+
+- All servers will reply `SendLastState` which includes their verifiable tip
+  header and the total difficulty of the chain.
+
+- The client will send `GetLastStateProof` to all servers which have replied
+  their last state. The `GetLastStateProof` messages are different for
+  different chain states, they are generated base on the sampling strategy
+  which the light client used.
+
+  _The sampling strategy is not a part of CKB. Each light client could
+  choose their own sampling strategy._
+
+  _In the official CKB light client, a sampling strategy based on the
```

Comment:
It's better to add a section about the official implementation and link to the Git repo.